validator/client/mock_validator.go

```diff
@@ -119,14 +119,8 @@ func (fv *FakeValidator) LogValidatorGainsAndLosses(_ context.Context, _ uint64)
 	return nil
 }
 
-// SaveProtections for mocking.
-func (fv *FakeValidator) SaveProtections(_ context.Context) error {
-	fv.SaveProtectionsCalled = true
-	return nil
-}
-
 // ResetAttesterProtectionData for mocking.
-func (fv *FakeValidator) ResetAttesterProtectionData() {
+func (fv *FakeValidator) ResetProtections(ctx context.Context) {
```

Comment:
Can we rename mock_validator.go to mock_validator_test.go? It shouldn't leak into production code.